### ISSUE TITLE
FusedLocationClient.java uses or overrides a deprecated API.

### ISSUE THREAD

**Issue by FabienZE** [Baseflow/flutter-geolocator#1204]
## 🐛 Bug Report

When build for debug, compiler says geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java uses or overrides a deprecated API.

Terminal output
```
Note: C:\Users\faher\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```

### Expected behavior

AssembleDebug in VSCODE 

**Version:** 4.1.7

**Platform:** 
- [ ] :iphone: iOS
- [X] :robot: Android

**Comment by battlecook** [comment]
same issue occurred.

**Comment by Kumar-71** [comment]
Same issue

**Comment by gommido** [comment]
I had this issue but  I realize that I forget to put the permissions in  AndroidManifest, when I added it, it works fine.
I don't know was that the real cause or not?

**Comment by Horned-Nonsense** [comment]
The solution is described in the [package](https://pub.dev/packages/geolocator).

Usage->Android

> On Android you'll need to add either the ACCESS_COARSE_LOCATION or the ACCESS_FINE_LOCATION permission to your Android Manifest. To do so open the AndroidManifest.xml file (located under android/app/src/main) and add one of the following two lines as direct children of the <manifest> tag (when you configure both permissions the ACCESS_FINE_LOCATION will be used by the geolocator plugin):
> 
`<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />`
`<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />`

> 
> Starting from Android 10 you need to add the ACCESS_BACKGROUND_LOCATION permission (next to the ACCESS_COARSE_LOCATION or the ACCESS_FINE_LOCATION permission) if you want to continue receiving updates even when your App is running in the background (note that the geolocator plugin doesn't support receiving and processing location updates while running in the background):
> 
 `<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />`

**Comment by onghuisheng** [comment]
same issue here

**Comment by Saipreetham0** [comment]
Same issue here also

**Comment by jangopper** [comment]
Same issue here but...

...not a train smash, no fix - but ONLY if 'deprecated' bothers you as much as it does me...

https://developers.google.com/android/guides/releases#october_13_2022

...a quick workaround ONLY to get rid of the 'geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java uses or overrides a deprecated API.' message:

Find (windows) X:\Users\xxxx\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java

Find (Line 85):
  private static LocationRequest buildLocationRequest(@Nullable LocationOptions options) {
    LocationRequest locationRequest = LocationRequest.create();

    if (options != null) {
      locationRequest.setPriority(toPriority(options.getAccuracy()));
      locationRequest.setInterval(options.getTimeInterval());
      locationRequest.setFastestInterval(options.getTimeInterval() / 2);
      locationRequest.setSmallestDisplacement(options.getDistanceFilter());
    }

    return locationRequest;
  }
  
  Replace with:
    private static LocationRequest buildLocationRequest(@Nullable LocationOptions options) {
      LocationRequest locationRequest = new LocationRequest.Builder(Priority.PRIORITY_HIGH_ACCURACY, 1000)
              .setWaitForAccurateLocation(false)
              .setMinUpdateIntervalMillis(500)
              .setMaxUpdateDelayMillis(1000)
              .build();

    return locationRequest;
  }
  
  No fix! Just a suggestion.

**Comment by ozaner** [comment]
> Replace with:
> ```
> private static LocationRequest buildLocationRequest(@nullable LocationOptions options) {
> LocationRequest locationRequest = new LocationRequest.Builder(Priority.PRIORITY_HIGH_ACCURACY, 1000)
> .setWaitForAccurateLocation(false)
> .setMinUpdateIntervalMillis(500)
> .setMaxUpdateDelayMillis(1000)
> .build();
> return locationRequest;
> ```

@jangopper This code basically fixes the problem so, instead of temporarily fixing it in your local package cache, you could make this a PR and fix it for everyone! (assuming the maintainers merge it).

**Comment by Horned-Nonsense** [comment]
@ozaner Why can't we use the solution from the package? I wrote about it above.

**Comment by ozaner** [comment]
> @ozaner Why can't we use the solution from the package? I wrote about it above.

I already had the `ACCESS_FINE_LOCATION` and the `ACCESS_BACKGROUND_LOCATION` in my manifest when I got the warning.

Although I don't seem to be getting the warning anymore, with or without the changes to the manifest, so I honestly don't know...

**Comment by Horned-Nonsense** [comment]
> > @ozaner Why can't we use the solution from the package? I wrote about it above.
> 
> I already had the `ACCESS_FINE_LOCATION` and the `ACCESS_BACKGROUND_LOCATION` in my manifest when I got the warning.
> 
> Although I don't seem to be getting the warning anymore, with or without the changes to the manifest, so I honestly don't know...

Idk if I did it right, but I added all 3 fields to the manifest and that fixed the problem for me.  Although, perhaps, 1 of the first two options was enough.

**Comment by Arcttyx** [comment]
Same here, and I already have <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" /> in the manifest

**Comment by haulinhen** [comment]
```
//	private static LocationRequest buildLocationRequest(@Nullable LocationOptions options) { 
 //	LocationRequest locationRequest = LocationRequest.create();
 
 // if (options != null) {
 //    locationRequest.setPriority(toPriority(options.getAccuracy()));
 //    locationRequest.setInterval(options.getTimeInterval());
 //    locationRequest.setFastestInterval(options.getTimeInterval() / 2);
 //    locationRequest.setSmallestDisplacement(options.getDistanceFilter());
  }
 // return locationRequest
 
 // Replace with:
	private static LocationRequest buildLocationRequest(@nullable LocationOptions options) {
		LocationRequest locationRequest = new LocationRequest.Builder(Priority.PRIORITY_HIGH_ACCURACY, 1000)
		.setWaitForAccurateLocation(false)
		.setMinUpdateIntervalMillis(500)
		.setMaxUpdateDelayMillis(1000)
		.build();
 
     return locationRequest;
 // End replace
  }
```
it's look like this after?

**Comment by danilocatalanoapsystem** [comment]
This warning comes out after doing flutter clean first and then flutter build apk. Even adding the configuration parameters in the androidManifest at least in my case, the warning message comes out. My fear is that from one day to the next geolocation doesn't work anymore and I can't publish.
Is there any way to fix this?
I saw that in 2020 the manufacturer created a new version that fixed the problem. Is it possible to have a version with the resolution of the problem?

**Comment by hanifmhd** [comment]
any update on this issue?

**Comment by mvanbeusekom** [comment]
This issue is solved with version 4.2.1 of the [geolocator_android](https://pub.dev/packages/geolocator_android/changelog) package. To update please run `flutter pub upgrade`.
